### PR TITLE
Analytics improvement

### PR DESCRIPTION
**Pull request by BLasan**
### Purpose
- fixes: https://github.com/wso2/api-manager/issues/2214

## Goals
Load API and Operation Policies of the APIs/API Products to the gateway and send the corresponding policies in the analytics event.

## Approach
1. Load API Policies and Operation Policies to the gateway on the deployment.
2. Fetch Operation and API Policies related to the the resource and the method on the API invocation.
3. To Enable/Disable Policies in the event, a separate configuration `enablePolicy` is introduced under `analytics` configuration in `deployment.toml` file. If this is enabled then operation and API policies will get loaded to the analytics event

```
[apim.analytics]
enable = true
auth_token = ""
enablePolicy = true
```

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.common.analytics/src/main/java/org/wso2/carbon/apimgt/common/analytics/publishers/dto/OperationPolicy.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (tharikaGitHub) [inline components/apimgt/org.wso2.carbon.apimgt.common.analytics/src/main/java/org/wso2/carbon/apimgt/common/analytics/publishers/dto/OperationPolicy.java:2]:
```suggestion
 * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```

File: components/apimgt/org.wso2.carbon.apimgt.common.analytics/src/main/java/org/wso2/carbon/apimgt/common/analytics/publishers/dto/URITemplate.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (tharikaGitHub) [inline components/apimgt/org.wso2.carbon.apimgt.common.analytics/src/main/java/org/wso2/carbon/apimgt/common/analytics/publishers/dto/URITemplate.java:2]:
```suggestion
 * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java [hunk 3]
```diff
@@ -1170,10 +1177,16 @@ private void attachURlMappingDetailsOfApiProduct(Connection connection, API api)
                 }
             }
         }
+
+        if(configs.containsKey(POLICY_ENABLED_FOR_ANALYTICS)) {
```
Comment (tharikaGitHub) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java:1181]:
```suggestion
        if (configs.containsKey(POLICY_ENABLED_FOR_ANALYTICS)) {
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java [hunk 4]
```diff
@@ -1260,6 +1273,112 @@ private void attachURLMappingDetails(Connection connection, String revisionId, A
...
```
Comment (tharikaGitHub) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java:1277]:
```suggestion
        if (configs.containsKey(POLICY_ENABLED_FOR_ANALYTICS)) {
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java [hunk 5]
```diff
@@ -973,6 +975,7 @@ private Permissions() {
 
     public static final String API_ANALYTICS = "Analytics.";
     public static final String API_USAGE_ENABLED = API_ANALYTICS + "Enabled";
+    public static final String API_ANALYTICS_POLICY_ENABLED = API_ANALYTICS + "PolicyEnabled";
```
Comment (tharikaGitHub) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java:978]: This constant does not seem to be used in any other place. May be we can remove.

**Review by npamudika — APPROVED** [review]

**Pull request merged**